framework/src/multiapps/FullSolveMultiApp.C

```diff
@@ -63,6 +68,20 @@ FullSolveMultiApp::restore(bool /*force*/)
     MultiApp::restore();
 }
 
+void
+FullSolveMultiApp::reinitialize()
+{
+  // Similarly to restore, which is used for the fixed-point iteration,
+  // we reinitialize our problem if it is requested by the user.
+  if (getParam<bool>("reinit_after_solve") && _has_an_app)
+    for (unsigned int i = 0; i < _my_num_apps; i++)
+    {
+      auto & app = _apps[i];
+      resetApp(i, app->getStartTime());
+      initialSetup();
```

Comment:
do we need this call?

Reply:
Yes, without the initialization it segfaults out due to uninitilized data.